kernel/kernel-api/src/main/java/io/delta/kernel/client/JsonHandler.java

```diff
@@ -40,17 +41,34 @@ public interface JsonHandler
     /**
      * Parse the given <i>json</i> strings and return the fields requested by {@code outputSchema}
      * as columns in a {@link ColumnarBatch}.
+     * <p>
+     * There are a couple special cases that should be handled for specific data types:
+     * <ul>
+     *    <li><b>FloatType and DoubleType:</b> handle non-numeric numbers encoded as strings
+     *    <ul>
+     *        <li>NaN: <code>"NaN"</code></li>
+     *        <li>Positive infinity: <code>"+INF", "Infinity", "+Infinity"</code></li>
+     *        <li>Negative infinity: <code>"-INF", "-Infinity""</code></li>
+     *    </ul>
+     *    </li>
+     *    <li><b>DateType:</b> handle dates encoded as strings in the format
+     *    <code>"yyyy-MM-dd"</code></li>
+     *    <li><b>TimestampType:</b> handle timestamps encoded as strings in the format
+     *    <code>"yyyy-MM-dd'T'HH:mm:ss.SSSXXX"</code></li>
```

Comment:
Is this the only format expected for the timestamp? I see in the tests you also have a few more different formats.

Reply:
This is the only format I could generate; I looked at [JacksonGenerator](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala) and tried playing with the SQL confs. XXX can be a zone offset or 'Z' for no offset

Reply:
Added to the project plan in the design doc to investigate this further later on